### ISSUE TITLE
Request: DISCARD ALL recycling method

### ISSUE THREAD

**Issue by jakajancar** [bikeshedder/deadpool#57]
How about a recycling method that uses [DISCARD ALL](https://www.postgresql.org/docs/12/sql-discard.html), guaranteeing a pristine connection?

**Comment by bikeshedder** [comment]
Great find. I didn't know this command existed. This should indeed be added. :+1:

**Comment by bikeshedder** [comment]
I just looked into this and found that `DISCARD ALL` might not actually be the best pick as it trashes the statement cache. I went with the following sequence of commands:

```sql
CLOSE ALL;
SET SESSION AUTHORIZATION DEFAULT;
RESET ALL;
UNLISTEN *;
SELECT pg_advisory_unlock_all();
DISCARD TEMP;
DISCARD SEQUENCES;
```

This is essentially `DISCARD ALL` but without `DEALLOCATE ALL` and `DISCARD PLAN` so that neither the prepared statements are deallocated nor the query plans are released.

**Comment by bikeshedder** [comment]
I just released `deadpool-postgres` 0.5.6 which includes this feature:

- https://github.com/bikeshedder/deadpool/blob/master/postgres/CHANGELOG.md
- https://crates.io/crates/deadpool-postgres/0.5.6
